### PR TITLE
#210: Resolves issues with update of Roles

### PR DESCRIPTION
**Pull request by DougMidgley**
# PR details

## What is the purpose of this pull request? (put an "X" next to an item)

_Please delete options that are not relevant._

- [ ] Documentation update
- [X] Bug fix
- [ ] New metadata support
- [ ] Enhanced metadata
- [ ] Add a CLI option
- [ ] Add something to the core
- [ ] Other, please explain:

## What changes did you make? (Give an overview)

Minor fix to ensure ObjectID is retrieved

## Is there anything you'd like reviewers to focus on?

General apporach

## Checklist

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] ESLint & Prettier are not outputting errors or warnings
- [X] README.md updated (if applicable)
- [NA] CHANGELOG.md updated
- [NA] ran `npm run docs` to update developer docs

### CODE REVIEW

**Comment by DougMidgley** [comment]
Give another check now. Found the cause and should be resolved (also had potential other impacts)

File: lib/metadataTypes/MetadataType.js [hunk 1]
```diff
@@ -340,18 +340,15 @@ class MetadataType {
             const metadataResults = createResults
                 .concat(updateResults)
                 .filter((r) => r !== undefined && r !== null)
-                .map((r) => r.body.Results)
+                .map((r) => r.Results)
```
Comment (JoernBerkefeld) [inline lib/metadataTypes/MetadataType.js:343]: was looking into this one as well when I tested. you sure there are no cases WITH `body.` ?

File: lib/metadataTypes/MetadataType.js [hunk 2]
```diff
@@ -340,18 +340,15 @@ class MetadataType {
...
             return this.parseResponseBody({ Results: metadataResults });
         } else {
             // put in Retrieve Format for parsing
             // todo add handling when response does not contain items.
             // @ts-ignore
-            const metadataResults = createResults
-                .concat(updateResults)
-                .filter((r) => r?.res?.body)
-                .map((r) => r.res.body);
+            const metadataResults = createResults.concat(updateResults).filter((r) => r);
```
Comment (JoernBerkefeld) [inline lib/metadataTypes/MetadataType.js:351]: filtering empty create or update lists here, right?

File: lib/metadataTypes/MetadataType.js (hunk 1 above)
Comment (JoernBerkefeld) [reply lib/metadataTypes/MetadataType.js:343]: in theory this impacts a lot:
image

Comment (JoernBerkefeld) [reply lib/metadataTypes/MetadataType.js:343]: the comment did say something about soap only but I wonder if your new SDK is doing the same for REST and SOAP here?

**Comment by JoernBerkefeld** [comment]
confirmed working for `roles` now.
checking other types...

**Review by JoernBerkefeld — APPROVED** [review]

**Pull request merged**